model.py

```diff
@@ -5646,8 +5652,63 @@ def update_availability(
             if self.work:
                 self.work.last_update_time = as_of
 
+        if changes_made:
+            message, args = self.circulation_changelog(
+                old_licenses_owned, old_licenses_available,
+                old_licenses_reserved, old_patrons_in_hold_queue
+            )
+            logging.info(message, *args)
+
         return changes_made
 
+    def circulation_changelog(self, old_licenses_owned, old_licenses_available,
+                              old_licenses_reserved, old_patrons_in_hold_queue):
+        """Generate a log message describing a change to the circulation.
+
+        :return: a 2-tuple (message, args) suitable for passing into 
+        logging.info or a similar method
+        """
+        edition = self.presentation_edition
+        message = 'CHANGED '
+        blah = object()
+        args = []
+        if edition:
+            message += '%s "%s" %s (%s)'
+            args.extend([edition.medium, 
+                         edition.title or "[NO TITLE]",
+                         edition.author or "[NO AUTHOR]",
+                         self.identifier]
+                    )
+        else:
+            message += '%s'
+            args.append(self.identifier)
+
+        def _part(message, string, old_value, new_value):
```

Comment:
You're not using message right now in this method, but you probably could be. You could also pass in `args` as well, and then get rid of lines 5693-4, 5698-9, 5703-4, and 5708-9. Either way works, though; it's up to you.